pkg/istio-agent/agent.go

```diff
@@ -308,100 +302,40 @@ func (a *Agent) initializeEnvoyAgent(ctx context.Context) error {
 	}
 	a.envoyAgent = envoy.NewAgent(envoyProxy, drainDuration, a.cfg.MinimumDrainDuration, localHostAddr,
 		int(a.proxyConfig.ProxyAdminPort), a.cfg.EnvoyStatusPort, a.cfg.EnvoyPrometheusPort, a.cfg.ExitOnZeroActiveConnections)
-	a.envoyWaitCh = make(chan error, 1)
 	if a.cfg.EnableDynamicBootstrap {
+		a.dynamicBootstrapWaitCh = make(chan error, 1)
 		// Simulate an xDS request for a bootstrap
-		a.wg.Add(1)
-		go func() {
-			defer a.wg.Done()
-
-			// wait indefinitely and keep retrying with jittered exponential backoff
-			backoff := 500
-			max := 30000
-		retries:
-			for {
-				// handleStream hands on to request after exit, so create a fresh one instead.
-				request := &bootstrapDiscoveryRequest{
-					node:        node,
-					envoyWaitCh: a.envoyWaitCh,
-					envoyUpdate: envoyProxy.UpdateConfig,
-				}
-				_ = a.xdsProxy.handleStream(request)
-				select {
-				case <-a.envoyWaitCh:
-					break retries
-				default:
+		// wait indefinitely and keep retrying with jittered exponential backoff
+		b := backoff.NewExponentialBackOff()
+		b.MaxElapsedTime = 30 * time.Second
```

Comment:
I think with non-zero MaxElapsedTime, after we hit max time the library starts returning -1 so we will have infinite tight loop.

IMO this library is too easy to use poorly, maybe we should create a wrapper in Istio pkg

Reply:
Let me take a close look at it. If so, it is necessary to wrap it